cmd/kubeless/function/deploy.go

```diff
@@ -79,9 +82,11 @@ var deployCmd = &cobra.Command{
 			logrus.Fatal(err)
 		}
 
-		if runtime != "" && !lr.IsValidRuntime(runtime) {
-			logrus.Fatalf("Invalid runtime: %s. Supported runtimes are: %s",
-				runtime, strings.Join(lr.GetRuntimes(), ", "))
+		if checkRuntime {
```

Comment:
this is fine but since `config` is initialized way above some other function may try to use that variable that may be `nil`. To make a bit more clear that the config is only used for the runtime check I would move the initialization here:

```
apiExtensionsClientset := kubelessUtils.GetAPIExtensionsClientOutOfCluster()
config, err := kubelessUtils.GetKubelessConfig(cli, apiExtensionsClientset)
if config != nil {
    lr := langruntime.New(config)
    lr.ReadConfigMap()
    if runtime ...
} else {
    logrus.Warnf("%v. Runtime check is disabled.", err)
}
```

Reply:
@andresmgot Thanks, please look at my changes